paper-ripple.html

```diff
@@ -116,362 +113,418 @@
+      },
+
+      get translationFraction () {
+        return Math.min(
+          1,
+          this.radius / this.containerMetrics.size * 2 / Math.sqrt(2)
+        );
+      },
+
+      get xNow () {
+        if (this.xEnd) {
+          return this.xStart + this.translationFraction * (this.xEnd - this.xStart);
         }
-        // else there is already a rAF
-      },
-
-      upAction: function() {
-        for (var i = 0; i < this.waves.length; i++) {
-          // Declare the next wave that has mouse down to be mouse'ed up.
-          var wave = this.waves[i];
-          if (wave.isMouseDown) {
-            wave.isMouseDown = false
-            wave.mouseUpStart = now();
-            wave.mouseDownStart = 0;
-            wave.tUp = 0.0;
-            break;
-          }
+
+        return this.xStart;
+      },
+
+      get yNow () {
+        if (this.yEnd) {
+          return this.yStart + this.translationFraction * (this.yEnd - this.yStart);
         }
-        this._loop && requestAnimationFrame(this._loop);
+
+        return this.yStart;
+      },
+
+      get isMouseDown () {
+        return this.mouseDownStart && !this.mouseUpStart;
       },
 
-      cancel: function() {
-        this.cancelled = true;
+      resetInteractionState: function () {
+        this.maxRadius = 0;
+        this.mouseDownStart = 0;
+        this.mouseUpStart = 0;
+
+        this.xStart = 0;
+        this.yStart = 0;
+        this.xEnd = 0;
+        this.yEnd = 0;
+        this.slideDistance = 0;
+
+        this.containerMetrics = new ContainerMetrics(this.element);
       },
 
-      animate: function(ctx) {
-        var shouldRenderNextFrame = false;
+      draw: function () {
+        var scale;
+        var translateString;
+        var dx;
+        var dy;
 
-        var deleteTheseWaves = [];
-        // The oldest wave's touch down duration
-        var longestTouchDownDuration = 0;
-        var longestTouchUpDuration = 0;
-        // Save the last known wave color
-        var lastWaveColor = null;
-        // wave animation values
-        var anim = {
-          initialOpacity: this.initialOpacity,
-          opacityDecayVelocity: this.opacityDecayVelocity,
-          height: ctx.height,
-          width: ctx.width
+        if (this.outerOpacity !== undefined) {
+          this.background.style.opacity = this.outerOpacity;
         }
 
-        for (var i = 0; i < this.waves.length; i++) {
-          var wave = this.waves[i];
+        this.wave.style.opacity = this.opacity;
 
-          if (wave.mouseDownStart > 0) {
-            wave.tDown = now() - wave.mouseDownStart;
-          }
-          if (wave.mouseUpStart > 0) {
-            wave.tUp = now() - wave.mouseUpStart;
-          }
+        scale = this.radius / (this.containerMetrics.size / 2);
+        dx = this.xNow - (this.containerMetrics.width / 2);
+        dy = this.yNow - (this.containerMetrics.height / 2);
+
+        translateString = 'translate3d(' + dx + 'px,' + dy + 'px,0)';
```

Comment:
There's a `translate3d` convenience function in base that help make this less verbose.